util/fibers/detail/scheduler.cc

```diff
@@ -200,8 +205,7 @@ Scheduler::~Scheduler() {
 }
 
 void Scheduler::Yield(FiberInterface* me) {
-  AddReady(me);
-  yield_occurred_ = true;
+  round_robin_run_.last_yield = true;
```

Comment:
Seems that it's simpler to introduce `Preempt(bool should_yield)` instead of having this state variable `last_yield`